include/SFML/Config.hpp

```diff
@@ -111,6 +111,14 @@
 #endif
 
 
+////////////////////////////////////////////////////////////
+// Ensure correct C++ language selection
+////////////////////////////////////////////////////////////
+#if (defined(_MSVC_LANG) && _MSVC_LANG < 201703L) || (!defined(_MSVC_LANG) && __cplusplus < 201703L)
+#error "C++17 required, enable C++17 for your compiler (e.g. -std=c++17 for GCC/Clang or /std:c++17 for MSVC)")
```

Comment:
```suggestion
#error "C++17 or newer required. Enable via -std=c++17 for GCC/Clang or /std:c++17 for MSVC"
```
Also note the removal of the trailing paren. That would probably cause issues.